File: 217. Contains Duplicate/contains_duplicate.py
"""
–––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
# 217. Contains Duplicate | LeetCode
Program | `contains_duplicate.py`

Daniel Bakas <daniel@semanytk.com>

Difficulty: Easy
Topics: Arrays

Given an integer array nums, return true if any value appears at least twice in
the array, and return false if every element is distinct.

Sep 19, 2022
–––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
"""

# * Imports
from typing import List


# * Main
def contains_duplicate(nums: List[int]) -> bool:
    return len(nums) != len(set(nums))


# * Tests
# Test 1
assert not contains_duplicate([1, 2, 3]), "Test 1: Failed"
print("Test 1: Passed")
# Test 2
assert contains_duplicate([1, 1]), "Test 2: Failed"
print("Test 2: Passed")
# Test 3
assert not contains_duplicate([1]), "Test 3: Failed"
print("Test 3: Passed")

File: 217. Contains Duplicate/containsDuplicate.js
/**
 * –––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
 * # 217. Contains Duplicate | LeetCode
 * Program | `containsDuplicate.js`
 *
 * Daniel Bakas <daniel@semanytk.com>
 *
 * Difficulty: Easy
 * Topics: Arrays
 *
 * Given an integer array nums, return true if any value appears at least
 * twice in the array, and return false if every element is distinct.
 *
 * Oct 18, 2022
 * –––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
 */

//* Main
containsDuplicate = (nums) => { return nums.length !== new Set(nums).size }

//* Tests
// Test 1
console.assert(!containsDuplicate([1, 2, 3]));
console.log("Test 1: Passed");
// Test 2
console.assert(containsDuplicate([1, 1]));
console.log("Test 2: Passed");
// Test 3
console.assert(!containsDuplicate([1]));
console.log("Test 3: Passed");